docs/dev/how-to/debug_permissions.mdx

```diff
@@ -0,0 +1,68 @@
+
+## Configure authorization for a code host connection
+
+1. Visit the Site admin settings page and select the Code host connections section. You should see a list of code host connections.
+![](https://storage.googleapis.com/sourcegraph-assets/Docs/how-to/debug-permissions/code_host_connections.png)
+2. Select the code host that you’d like to view, which should show you its configuration page.
+![](https://storage.googleapis.com/sourcegraph-assets/Docs/how-to/debug-permissions/gitlab_connection.png)
+    - Here you can see that this GitLab connection is configured with authorization. If the authorization field is missing completely then the code host connection will be marked as “unrestricted” and all repositories synced by that code host connection will be accessible to everyone on Sourcegraph.
+    - Not all authorization configurations are the same. Visit the code hosts docs page for more detailed information on how to configure authorization for the specific code host you’re dealing with.
+3. Once you’ve made adjustments and saved the config, a sync should be triggered for the code host. If no sync was triggered, hit the Trigger manual sync button.
+4. Next, check the user permissions screen to see if things are behaving as expected now, and if nothing has changed, try scheduling a permissions sync.
+
+## Is the repo public on the code host but not public on Sourcegraph or vice versa?
+
+1. Confirm that the repository on the code host has the visibility you expect it to have.
+    - Does the code host even have a concept of visibility? Some enterprise code hosts don’t have a concept of “publically available” repositories.
+2. Confirm that the visibility of the repository on Sourcegraph differs from what you saw on the code host itself.
+![](https://storage.googleapis.com/sourcegraph-assets/Docs/how-to/debug-permissions/repo_search.png)
+    - Private repositories on Sourcegraph will have the Private label as displayed above. If that label is not there, consider the repo to be public.
+3. If there is a discrepancy between the visibility of the repository on the code host vs. Sourcegraph, it could point to a bug on Sourcegraph’s side.
+
+## Is the repository actually present on Sourcegraph?
+
+1. Navigate to the **Site admin** settings and select **Repositories** under the **Repositories** section in the navigation menu.
+![](https://storage.googleapis.com/sourcegraph-assets/Docs/how-to/debug-permissions/repositories_option.png)
+2. Search for the repository in question.
+![](https://storage.googleapis.com/sourcegraph-assets/Docs/how-to/debug-permissions/repo_search.png)
+3. If the repository does not show up, but you expect it to, the instance may be configured to enforce repository permissions even for site admins. In this case, the site admin will not be able to see the repository unless they have explicit access to the repository. It may be worth temporarily disabling this enforcement for debugging purposes.
```

Comment:
```suggestion
3. If the repository does not show up, but you expect it to, the instance may be configured to enforce repository permissions even for site admins. In this case, the site admin can only see the repository if they have explicit access. It may be worth temporarily turning off this enforcement for debugging purposes.
```